lib/core.ex

```diff
@@ -7,8 +7,18 @@ defmodule Core do
 
   def create_playlist(channel_id) do
     video = Youtube.get_channel_last_video(channel_id)
-    Line.send_text_message("A new video was published #{video.video_title}")
-    create_playlist_from_video(video.video_id)
+
+    case video do
+      {:ok, %{video: video}} ->
```

Comment:
I think you could have `{:ok, video}` here, and `{:error, :no_video}`... or just return a list directly and pattern-match here on `[] | [video] | videos`?